src/sentry/api/base.py

```diff
@@ -213,14 +218,28 @@ def permission_denied(self, request, message=None, code=None):
         the appropriate exception according to parent DRF function.
         """
         permissions = self.get_permissions()
+        if request.user.is_authenticated and len(permissions) == 1:
+            permission_cls = permissions[0]
+            use_staff = features.has("auth:enterprise-staff-cookie", actor=request.user)
+
+            # TODO(schew2381): Remove SuperuserOrStaffFeatureFlaggedPermission
+            # from isinstance checks once feature flag is removed.
+            can_be_superuser = request.user.is_superuser
+            has_only_superuser_permission = isinstance(
+                permission_cls, (SuperuserPermission, SuperuserOrStaffFeatureFlaggedPermission)
+            )
 
-        can_be_superuser = request.user.is_authenticated and request.user.is_superuser
-        has_only_superuser_permission = len(permissions) == 1 and isinstance(
-            permissions[0], SuperuserPermission
-        )
+            if not use_staff and can_be_superuser and has_only_superuser_permission:
+                raise SuperuserRequired()
+
+            can_be_staff = request.user.is_authenticated and request.user.is_staff
+            has_only_staff_permission = isinstance(
+                permission_cls, (StaffPermission, SuperuserOrStaffFeatureFlaggedPermission)
```

Comment:
maybe add a note that if you need superuser OR staff (`SuperuserOrStaffFeatureFlaggedPermission`), for now we are only hitting this part if you have the feature flag in the first place?

Reply:
This permission is for staff to replace superuser entirely rather than being an OR between the two. We need the feature flag permission for the transition, and then we'll delete this feature flag entirely and replace it with `StaffPermission`

Reply:
ok see i totally misunderstood -- i think a note would be helpful 😅